### PR TITLE
update model format description

### PR DESCRIPTION
**Pull request by sunya-ch**
This PR added the information about model format for future contribution corresponding to the issue https://github.com/sustainable-computing-io/kepler-model-server/issues/63.

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

### CODE REVIEW

File: doc/model_format.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+<!-- vscode-markdown-toc -->
+- [Model Format](#model-format)
+  - [Metadata](#metadata)
+  - [Model Content](#model-content)
+    - [1. Model Weight for Linear Regression](#1-model-weight-for-linear-regression)
+    - [2. Archieved Model](#2-archieved-model)
+    - [3. Component Model](#3-component-model)
+      - [3.1. Model Weight](#31-model-weight)
+      - [3.2. Archieved Model](#32-archieved-model)
```
Comment (KaiyiLiu1234) [inline doc/model_format.md:9]: Archieved should be Archived

File: doc/model_feature.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+*Note:* WORKLOAD_FEATURES includes COUNTER_FEAUTRES, CGROUP_FEATURES, IO_FEATURES, BPF_FEATURES and KUBELET_FEATURES.
+
+**List of exported metrics for each feature:**
+Queries: node_energy_stat, pod_energy_stat
+[metric declaration](https://github.com/sustainable-computing-io/kepler/blob/c8f0528980944d3952f001ac09cd4e285cebe73e/pkg/collector/prometheus_collector.go#L40-L80)
+
+Features|Kepler exporting labels in `pod_energy_stat` metric
+---|---
+COUNTER_FEAUTRES| - cache_miss<br>- cpu_cycles<br>- cpu_instr
```
Comment (KaiyiLiu1234) [inline doc/model_feature.md:21]: Correct me if incorrect, but shouldn't cpu frequency also be included?

File: doc/model_format.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	Variance float64 `json:"variance"`
+	Weight   float64 `json:"weight"`
+}
+```
+
+###  2. <a name='ArchievedModel'></a>Archieved Model
+The model file can be saved in multiple format depends on model class. The model will be applied by [genearl estimator sidecar](https://github.com/sustainable-computing-io/kepler-estimator). The sidecar is called by [Kepler EstimatorSidecarConnector](https://github.com/sustainable-computing-io/kepler/blob/main/pkg/model/estimator/sidecar/estimate.go#L83).
+
+- scikit: the model is saved in `.sav`.
+- keras: the model is saved as TensorFlow SavedModel format (folder).
```
Comment (KaiyiLiu1234) [inline doc/model_format.md:91]: model is saved as Tensorflow SavedModel format (folder) and exported as an archived zip file.

File: doc/model_output.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+For each kind, the model can be in four following forms.
+
+Key Name| Model Form|Output per single input|
+---|---|---|
+Power|achived model|single power value|
+ModelWeight|weights for normalized linear regression model|single power value|
+ComponentPower|achived model|multiple power values for each components (e.g., core, dram)
+ComponentModelWeight|weights for normalized linear regression model|multiple power values for each components (e.g., core, dram)|
+
+The achived model allows more choices of learning approachs to estimate the power which can expect higher accuracy and overfitting resolutions while weights of linear regression model provides simplicity and lightweight. **Accordingly, the achived model is suitable for the systems that run intensive and non-deterministic workloads which require accurate prediction while the weights of linear regression is suitable for the systems which run for deterministic workloads in general.**
```
Comment (KaiyiLiu1234) [inline doc/model_output.md:22]: Do you mean archived? What is achived model?

**Review by KaiyiLiu1234 — CHANGES_REQUESTED** [review]
For specific changes refer to the comments. I have also summed up the changes recommended in the following. In README.md, change learning rate from 0.5 to 0.01. This was decreased for online learning. In model_feature.md, I think it is worth including cpu_frequency in counter features since that is now being used in the node model training (PR to be added today). . In model_format.md, Archieved should be Archived. Also note that for keras, models are saved as SaveModel (folder) but exported as an archived zip file I believe.

File: doc/model_format.md (hunk 3 above)
Comment (sunya-ch) [reply doc/model_format.md:91]: I zip the top folder of the model including the metadata for both scikit and keras in the archive-type model. Will add that line.

File: doc/model_feature.md (hunk 2 above)
Comment (sunya-ch) [reply doc/model_feature.md:21]: It should be the system feature not the counter feature. The node training should require the full feature including the system feature and the rest.

File: doc/model_output.md (hunk 4 above)
Comment (sunya-ch) [reply doc/model_output.md:22]: Yes. I will put the link to the format page.

**Comment by sunya-ch** [comment]
> For specific changes refer to the comments. I have also summed up the changes recommended in the following. In README.md, change learning rate from 0.5 to 0.01. This was decreased for online learning. In model_feature.md, I think it is worth including cpu_frequency in counter features since that is now being used in the node model training (PR to be added today). . In model_format.md, Archieved should be Archived. Also note that for keras, models are saved as SaveModel (folder) but exported as an archived zip file I believe.

Thank you for the comments and reviews. I made the changes in README, edited the typo, and added the line to state the zip format. For the feature, I would like to still separate CPU frequency from the counter because in some system such as on VM that is bypassed the hardware counter the CPU frequency is not also bypassed. I kept it as a system feature.

**Comment by sunya-ch** [comment]
For more clarification, I also updated the model_feature page to list the kepler metric linking it to the features and labels. I just noticed that the prom query for the online training needs to be updated with this new query list too.

**Comment by KaiyiLiu1234** [comment]
LGTM. I will open issues in the event some things can be changed or make a PR. Thanks!

**Review by KaiyiLiu1234 — APPROVED** [review]
